### PR TITLE
Remove unused getRuntimePodStatus func from Kubelet

### PR DESCRIPTION
**Pull request by pmorie**
@dchen1107 @vishh

Found while debugging #20403

### CODE REVIEW

File: pkg/kubelet/kubelet.go [hunk 1]
```diff
@@ -3137,18 +3137,18 @@ func GetPhase(spec *api.PodSpec, info []api.ContainerStatus) api.PodPhase {
...
-	start := kl.clock.Now()
-	defer func() {
-		metrics.PodStatusLatency.Observe(metrics.SinceInMicroseconds(start))
-	}()
-	if kl.podCache != nil {
-		return kl.podCache.Get(pod.UID)
-	}
-	return kl.containerRuntime.GetPodStatus(pod.UID, pod.Name, pod.Namespace)
-}
+// // Get the internal PodStatus from the cache if the cache exists;
```
Comment (vishh) [inline pkg/kubelet/kubelet.go]: Were you intending to delete or comment out?

Comment (pmorie) [inline pkg/kubelet/kubelet.go]: @vishh intending to delete, and thought I had deleted it.  ETOOMANYTHINGS

**Comment by vishh** [comment]
LGTM. Thanks for the cleanup @pmorie :bow:

**Comment by pmorie** [comment]
lol @ bow

**Pull request merged**